### PR TITLE
Feature set allowed commands in runtime

### LINKED ISSUE #2: Be able to filter allowed command in each submission
Requests should support an `allowed_commands` settings that describes which commands are available. 

For example: 

```json
{
  "query": "cd ..",
  "settings": { 
     "allowed_commands": ["cd", "ls", "touch"]
  }
}
```

### PR DESCRIPTION
**Pull request by julian-berbel**
Fixes #2.

~I'll add tests before formally opening this but~ this is basically it :smile_cat:

### CODE REVIEW

File: lib/try_hook.rb [hunk 1]
```diff
@@ -70,4 +71,15 @@ def create_tempfile
...
 
+  def required_commands
+    %w(bash sh)
+  end
+
+  def default_allowed_commands
+    %w(whoami cat cp grep ln ls mkdir mv rm rmdir sed touch git head tail wc)
+  end
+
+  def allowed_commands
```
Comment (luchotc) [inline lib/try_hook.rb]: I would leave allowed_commands as
`@custom_allowed_commands.presence || default_allowed_commands`

and 

¿available_commands? as `[allowed_commands, required_commands].join " "`

just for clarity

**Review by luchotc — CHANGES_REQUESTED** [review]
It's your Mona Lisa :stuck_out_tongue: 
In addition to that minor detail, I'm thinking if we shouldn't provide a bounded list of possible commands. I'm a bit concern about any security issues that we could have if we leave the list unrestricted.

Maybe we could have something like a list of total commands which can be exposed. It's would also helpful in case we want to show that list in an UI :stuck_out_tongue:

File: lib/try_hook.rb (hunk 1 above)
Comment (julian-berbel) [reply lib/try_hook.rb]: sounds reasonable!

**Comment by julian-berbel** [comment]
> In addition to that minor detail, I'm thinking if we shouldn't provide a bounded list of possible commands. I'm a bit concern about any security issues that we could have if we leave the list unrestricted.
> 
> Maybe we could have something like a list of total commands which can be exposed. It's would also helpful in case we want to show that list in an UI

Sounds great!

Of course that would mean if we ever want to expand the list we'll have to release a new version of the runner, but I'm all in for security measures!

We should discuss what that list should be, too.!

File: lib/try_hook.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def tempfile_extension
   end
 
   def compile_file_content(r)
+    @custom_allowed_commands = r.settings.try { |settings| settings['allowed_commands'] }
```
Comment (flbulgarelli) [inline lib/try_hook.rb]: Perhaps we could delegate this line

**Comment by julian-berbel** [comment]
Before closing we should look into this: mumuki/mumukit#102.

**Review by luchotc — APPROVED** [review]
Beautiful :kissing_heart: 
If you are merging it now, please open an issue so we don't forget to expose the settings list when we do https://github.com/mumuki/mumukit/issues/102

**Review by carobaldino — APPROVED** [review]

**Pull request merged**